### ISSUE TITLE
Questions about Base64 encoded data for exampleImageIdLoader.js

### ISSUE THREAD

**Issue by mtm12** [cornerstonejs/cornerstone#504]
Is the image108Base64 example data encoded RGB or Hounsfield values?

**Comment by chafey** [comment]
Neither - it is the 16 bit pixel data (7fe0,0010) extracted from a DICOM
file.  You probably do NOT want to encode your data like this though, use
the cornerstoneWADOImageLoader

**Comment by mtm12** [comment]
Hi Chris. Thank you for the response.
I'm looking into extracting a user-specified volume from a DICOM stack loaded into the OHIF viewer and displaying it in a separate window before doing further computational image processing. I'm not sure the best way to get the volume. Could I extract the 16 bit pixel data from the cornerstoneTools.js and then loading it using cornerstoneWADOImageLoader? What variable is the volume store in, in cornerstoneTools.js?

**Comment by chafey** [comment]
You can extract pixels from images using cornerstone core and then build
your own volume:

https://docs.cornerstonejs.org/advanced/retrieving-pixel-data.html

**Comment by mtm12** [comment]
I actually use that function for a 2D computational image process. Is there a way to get 3D data from the OHIF element? getPixels() returns a 2D array. Is there a way to get 3D volume pixels by specifying x,y,z?

**Comment by chafey** [comment]
No, cornerstone is a 2d library, it has no 3d functionality.  OHIF uses
vtkjs for MPR/VR - maybe look there?

**Comment by mtm12** [comment]
I can use getPixel() to get the pixels for the current slice but how do I get the pixels for the previous slice or the next slice to build the volume?

**Comment by chafey** [comment]
You can get the Image object by calling loadImage() or loadAndCacheImage()
with the imageId you want pixel data from

https://docs.cornerstonejs.org/api.html#image
https://docs.cornerstonejs.org/api.html#loadimage
https://docs.cornerstonejs.org/api.html#loadandcacheimage

The image object has a function getPixelData() which will return the
underlying stored pixel data from the DICOM SOP Instance

Chris
